.gitlab/deploy/runner/kubernetes/deploy.sh

```diff
@@ -33,24 +33,63 @@ if [ -z "${runner_registration_token}" ]; then
   exit 1
 fi
 
-kubectl apply -f - <<EOF
-apiVersion: rbac.authorization.k8s.io/v1
-kind: ClusterRoleBinding
-metadata:
-  name: gitlab-runner:cluster-admin
-subjects:
-- kind: ServiceAccount
-  name: default
-  namespace: ${namespace}
-roleRef:
-  kind: ClusterRole
-  name: cluster-admin
-  apiGroup: rbac.authorization.k8s.io
-EOF
-
 helm upgrade "${installation_name}" gitlab/gitlab-runner \
   --install \
+  --wait \
   --namespace "${namespace}" \
   --values .gitlab/deploy/runner/kubernetes/values.yaml \
   --set "runners.namespace=${namespace}" \
   --set "runnerRegistrationToken=${runner_registration_token}"
+
+# shellcheck disable=SC1004,SC2016
+kubectl patch "configmap/${installation_name}-gitlab-runner" \
+  --namespace "${namespace}" \
+  --patch '
+"data":
+  "entrypoint": |
+    #!/bin/bash
+    set -e
+    mkdir -p /home/gitlab-runner/.gitlab-runner/
+    cp /scripts/config.toml /home/gitlab-runner/.gitlab-runner/
+
+    # Register the runner
+    if [[ -f /secrets/accesskey && -f /secrets/secretkey ]]; then
+      export CACHE_S3_ACCESS_KEY=$(cat /secrets/accesskey)
+      export CACHE_S3_SECRET_KEY=$(cat /secrets/secretkey)
+    fi
+
+    if [[ -f /secrets/gcs-applicaton-credentials-file ]]; then
+      export GOOGLE_APPLICATION_CREDENTIALS="/secrets/gcs-applicaton-credentials-file"
+    else
+      if [[ -f /secrets/gcs-access-id && -f /secrets/gcs-private-key ]]; then
+        export CACHE_GCS_ACCESS_ID=$(cat /secrets/gcs-access-id)
+        # echo -e used to make private key multiline (in google json auth key private key is oneline with \n)
+        export CACHE_GCS_PRIVATE_KEY=$(echo -e $(cat /secrets/gcs-private-key))
```

Comment:
Please prefer `printf` over `echo -e`; in some shells `echo` always interprets escapes and prints `-e` literally. (Not an issue here, since we always use `bash`, but still…)
```suggestion
        export CACHE_GCS_PRIVATE_KEY="$(printf "%b" "$(cat /secrets/gcs-private-key)")"
```

Reply:
Same as above.